### ISSUE TITLE
Support for Remix

### ISSUE THREAD

**Issue by Marcisbee** [elysiajs/elysia#12]
Any plans to support Remix with [custom server adapter](https://remix.run/docs/en/v1/other-api/adapter) similar to `@remix-run/express`?

**Comment by SaltyAom** [comment]
Sure, I love Remix.

*Added to roadmap*

**Comment by SaltyAom** [comment]
Blocking on: https://github.com/oven-sh/bun/issues/1864

**Comment by Marcisbee** [comment]
Is this unblocked via this PR?

- https://github.com/oven-sh/bun/pull/2019

**Comment by SaltyAom** [comment]
Likely yes, will continue working on an adapter after the PR landed on stable (likely Bun 0.5.7)

**Comment by tobimori** [comment]
any update? :)

**Comment by rashidtvmr** [comment]
any update?

**Comment by wladiston** [comment]
Apparently  elysia + remix is already supported. You just need to make some tweeks
https://github.com/wladiston/elysia-remix

**Comment by TaQuanMinhLong** [comment]
I did also manage to make it work, though only with production mode

```ts
// plugins/remix.ts
import type { ServerBuild } from "@remix-run/node";
import { createRequestHandler } from "@remix-run/node";
import { join, parse } from "path";
import { Elysia } from "elysia";

export type RemixPlugin<T extends ServerBuild = any> = (
  serverBuild: T,
  publicDir?: string
) => Elysia;

export const remix: RemixPlugin = (serverBuild, publicDir = join(process.cwd(), "public")) =>
  new Elysia().all("*", async function ({ request, store, path, set }) {
    if (request.method === "GET" && parse(path).ext) {
      const filepath = join(publicDir, path);
      const file = Bun.file(filepath);
      const fileExist = await file.exists();
      if (!fileExist) {
        set.status = 404;
        return { error: "Requested resource does not exist" };
      }
      set.headers["Cache-Control"] = "public,max-age=31536000,immutable";
      return file;
    }
    /**
     * This only works for production build
     *
     * For development use remix dev instead
     */
    const handleRequest = createRequestHandler(serverBuild, "production");
    return await handleRequest(request, store);
  });
```

```ts
// server.ts
import { remix, logger } from "./plugins";
import { Elysia } from "elysia";
import * as build from "./build/index";

const port = process.env.PORT;

const app = new Elysia()
  .state("appLoadContext", "test")
  .use(logger())
  .use(remix(build))
  .listen(port ? Number(port) : 3000);

console.log(`Server is running at http://${app.server?.hostname}:${app.server?.port}`);
```

```ts
// Axios accept-encoding header should be set to gzip to work
...
headers: {
    common: {
      "Accept-Encoding": "gzip",
    },
  },
...
```

I'n not sure about this but after few clicks to test the router link, i feel that running original remix with bun is still somehow smoother than running remix by elysia 😂 It could be my implementation broke somewhere

**Comment by sabinpuiu** [comment]
Any update for official remix support?

**Comment by Danieljs-codes** [comment]
Any update for official remix support 🥹

**Comment by sigmachirality** [comment]
+1

**Comment by Danieljs-codes** [comment]
+1

**Comment by stephen776** [comment]
Is there any current path here that works now that remix uses vite?

**Comment by broisnischal** [comment]
do we have any example use of elysia and remix as monorepo?

**Comment by dy0gu** [comment]
Can confirm, @TaQuanMinhLong's solution worked in production only. 

It would be great to have official Remix support, elysia is looking amazing and I have already replaced my backend servers. 

I'm just missing it on the frontend for that sweet consistency now!

**Comment by kravetsone** [comment]
https://github.com/user-attachments/assets/725a4a85-d883-4a62-af81-9335482d9d2c

I work on it

SOON...

**Comment by kravetsone** [comment]
I publish [elysia-remix@0.0.1](https://github.com/kravetsone/elysia-remix)!

Currently, sending files from public dir in development mode does not work 

Waiting for your stars and issues ✨

**Comment by kravetsone** [comment]
@Marcisbee Can you share the feedback and if everything is fine close the issue?

**Comment by dy0gu** [comment]
Hey @kravetsone, I tried out your package and it seems pretty stable. 

I left a Issue + PR to fix some of the problems in the example you give but other than that it seems great! 

However, I think a working `public` directory in development is essential and we should make it a priority to fix that before closing this issue.

**Comment by kravetsone** [comment]
> Hey @kravetsone, I tried out your package and it seems pretty stable. 
> 
> I left a Issue + PR to fix some of the problems in the example you give but other than that it seems great! 
> 
> However, I think a working `public` directory in development is essential and we should make it a priority to fix that before closing this issue. I'd like to help but have no idea where to start so I guess I'll open another Issue on your repo about it and we'll go from there.

But `public` folder works today
Do u have issues with it?

**Comment by kravetsone** [comment]
> > Hey @kravetsone, I tried out your package and it seems pretty stable. 
> > 
> > I left a Issue + PR to fix some of the problems in the example you give but other than that it seems great! 
> > 
> > However, I think a working `public` directory in development is essential and we should make it a priority to fix that before closing this issue. I'd like to help but have no idea where to start so I guess I'll open another Issue on your repo about it and we'll go from there.
> 
> But `public` folder works today
> Do u have issues with it?

Ahh really i forgot to mention it here

**Comment by dy0gu** [comment]
That's weird, I tried it today with the lastest version of Bun, Elysia itself and your package. I got the exact opposite of that behavior.

I created a [minimal repository](https://github.com/dy0gu/elysia-remix-bug-demo) so you can see if you're able to reproduce the issue.

This is what happens (quoted from the  README.md inside that repository):

> Tested with Bun version 1.1.21
> 
> - Install dependencies with Bun
> 
> ```shell
> bun install
> ```
> 
> - Run in development mode:
> 
> ```shell
> bun run dev
> ```
> 
> **In this mode, all pages work normaly but files in the public folder are not being served.**
> 
> - Run in production mode:
> 
> ```shell
> bun run build && bun run start
> ```
> 
> **In this mode, all files in the public folder are being served but every page returns "NOT_FOUND"**

Side note: I tried with and without the `app` folder inside `src`, changing the corresponding `appDirectory` in the Remix `vite.config.ts` plugin config.

**Comment by SaltyAom** [comment]
Closing as supported as 3rd party http://github.com/kravetsone/elysia-remix

**Comment by bryanltobing** [comment]
not sure if i should make a new issue for this question. but is elysia already supported with tanstack start api routes?
